CMakeLists.txt

```diff
@@ -9,6 +9,21 @@ project(opentelemetry-cpp)
 # Mark variables as used so cmake doesn't complain about them
 mark_as_advanced(CMAKE_TOOLCHAIN_FILE)
 
+if(DEFINED ENV{ARCH})
+  # Architecture specified manually via env var, e.g. ARM64
+  set(ARCH $ENV{ARCH})
+else()
+  # Architecture detection logic for Intel chipsets only. For other
+  # architectures - please specify manually.
+  if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+    # 64 bits
+    set(ARCH x64)
+  elseif(CMAKE_SIZEOF_VOID_P EQUAL 4)
```

Comment:
Also check target is Intel?

Reply:
Yes, I agree we should.. My point was mostly that if someone builds for non-Intel (arm, risc, ppc), they'd have to specify the `ARCH` variable manually. Since this variable is currently only being used for Windows dependencies and/or with vcpkg, possible supported architectures are `[x86, x64, arm, arm64]`. I can try to improve this to handle Intel and ARM, but with a disclaimer above that any other non-Intel and non-ARM architectures should go thru `if(DEFINED ENV{ARCH})` path above. Let me think it over.

Reply:
@ThomsonTan - I carefully reviewed the options. The challenge is that on Windows - CMake does not necessarily properly populate the target architecture. This is described in CMake documentation. Best we can come up with:

- assume that users know what arch they are compiling. And if this is ARM - on Windows they must specify the target using environment variable, such as `set ARCH=arm` or `set ARCH=arm64`  , which will then be respected by the rest of logic that installs dependencies via `vcpkg`.

- if users don't specify, most likely they are building for Intel or ARM, nothing exotic.. And generic arch lookup would look like this (bulky!). This is much smarter than what I had done before, but doesn't really add much additional benefit.. 

```cmake

if(DEFINED ENV{ARCH})
  # Architecture may be specified via ARCH environment variable
  set(ARCH $ENV{ARCH})
else()
  # Autodetection logic that populates ARCH environment variable
  if(CMAKE_SYSTEM_PROCESSOR MATCHES "amd64.*|x86_64.*|AMD64.*")
    # Windows may report AMD64 even if target is 32-bit
    if(CMAKE_SIZEOF_VOID_P EQUAL 8)
      # 64 bits
      set(ARCH x64)
    elseif(CMAKE_SIZEOF_VOID_P EQUAL 4)
      # 32 bits
      set(ARCH x86)
    endif()
  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "i686.*|i386.*|x86.*")
    set(ARCH x86)
  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^(aarch64.*|AARCH64.*|arm64.*|ARM64.*)")
    set(ARCH arm64)
  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^(arm.*|ARM.*)")
    set(ARCH arm)
  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^(powerpc|ppc)64le")
    set(ARCH ppc64le)
  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^(powerpc|ppc)64")
    set(ARCH ppc64)
  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^(mips.*|MIPS.*)")
    set(ARCH mips)
  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^(riscv.*|RISCV.*)")
    set(ARCH riscv)
  else()
    message(FATAL_ERROR "opentelemetry-cpp: unrecognized target processor configuration!")
  endif()
endif()
message(STATUS "Building for architecture: ARCH=${ARCH}")

```

The only benefit is if we ever want to auto-install dependencies via `vcpkg` on Linux or Mac, for example, then knowing the target triplet comes in handy. It's a bit of an overkill, and I don't see a simpler solution. I looked at how Samsung did in in their repo, and for Windows they just assume that users manually specify the target arch by parameter.